### DISCUSSION TITLE
Reindexing a non blank library doesn't eliminate old data from library/originals

### DISCUSSION THREAD

**Discussion by TheodorDumitru** [photoprism/photoprism#4473] [Q&A]
Reindexing a non blank library doesn't eliminate old data from library/originals

---------------------------------------------------------------------------

#### 1. What is not working as documented?

I added elements to my library to test things, i deleted everything and then added some more in my process of testing photoprism and if i go under library to originals, i can access the previous folders I had created initially even if they are empty. They have the thumbnail as well, but I just can't delete them. The only options I get if I select any of them are to add to another album or download but because they don't exist, both actions fail

#### 2. How can we reproduce it?

1. Add photos and albums
2. Delete everything 
3. Add something entirely different
4. Reindex all originals with complete rescan and cleanup selected (or not, I tried every combination)
5. Check library\originals not having just your last upload but also a history of previous elements (after the last test I have 1 picture in my library, in a folder I created)![Screenshot_20240821_051916_Firefox.jpg](https://github.com/user-attachments/assets/e22b7afc-e08f-459c-beec-313eeae55f4e)

Here are my settings. General settings are all checked. From the advanced I activated logs and then tried reindexing with all thr backup checked and then unchecked, same result.  
- ![Screenshot_20240821_052603_Firefox.jpg](https://github.com/user-attachments/assets/6979efbe-6e42-4cf2-9d9a-9bd13a39c168)

![Screenshot_20240821_052533_Firefox.jpg](https://github.com/user-attachments/assets/e361271b-ea9e-4452-a5dd-225d81fcdcc1)

![Screenshot_20240821_052518_Firefox.jpg](https://github.com/user-attachments/assets/740b2b5b-c9e9-42cc-9fac-fb7a05770f46)

![Screenshot_20240821_052632_Firefox.jpg](https://github.com/user-attachments/assets/49d6c1c1-453b-4fd9-a69f-a5a354a18231)

![Screenshot_20240821_052617_Firefox.jpg](https://github.com/user-attachments/assets/e99b7075-a388-48fd-bdbf-ff630713d469)

![Screenshot_20240821_052527_Firefox.jpg](https://github.com/user-attachments/assets/1ac855d8-371e-448d-bff5-0a360149fd34)

![Screenshot_20240821_052107_Firefox.jpg](https://github.com/user-attachments/assets/e2f23b33-0090-4fb8-8140-e63ab8373310)

![Screenshot_20240821_051945_Firefox.jpg](https://github.com/user-attachments/assets/a290702e-e965-49a6-a55e-31b38f63a334)

![Screenshot_20240821_051951_Firefox.jpg](https://github.![Screenshot_20240821_052117_Firefox.jpg](https://github.com/user-attachments/assets/4930757d-6a73-4de7-819a-9fc11b227811)

c31cbfa)

![Uploading Screenshot_20240821_052117_Firefox.jpg …]()

![Screenshot_20240821_052058_Firefox.jpg](https://github.com/user-attachments/assets/092d042b-2d56-4b76-b64c-0222eac9df2c)

![Screenshot_20240821_051936_Firefox.jpg](https://github.com/user-attachments/assets/ac841af6-c705-4619-bdcc-0607d30f25c5)

#### 3. What behavior do you expect?

I expect a clean originals without my history. I delete photos for a reason. They're not in my face so at least I can hide them from my main view, but there is a ghost verion of them and I didn't expect that 

#### 4. What could be the cause of your problem?

I think it might be in the compose.yml file or settings something really important that I'm missing and just couldn't find any information about that

#### 6. Which software versions do you use?

(a) PhotoPrism Architecture & Build Number: AMD64 (x86 processor), build: 240711-2197af848 

(b) Database Type & Version: MariaDB 11.4

(c) Operating System Types & Versions: open media vault 7, based on debian (to run photoprism in a docker compose) 

To access: windows 10 via Chrome or Firefox + android 14 via Firefox, no extensions or plugins anywhere

#### 7. On what kind of device is PhotoPrism installed?

(a) Device / Processor Type: laptop with Intel core i7 4510U

(b) Physical Memory & Swap Space in GB: 4

(c) Storage Type: 250 gb ssd sata

**Comment by lastzero** [comment]
- https://github.com/photoprism/photoprism/discussions/2460
- https://docs.photoprism.app/user-guide/organize/delete/#delete-all-archived-photos

**Comment by TheodorDumitru** [comment]
That didn't solve my situation. After I ran the commands, at a first glance, the library seemed empty but it still had the ghost folders and thumbnails in library/originals. Then I tried to reindex the app again, both check boxes checked, and boom, I'm right where I was before running the commands. My photo came back in my album + the ghosts as well. Any other idea I can try?

**Comment by lastzero** [comment]
As documented, the reset command does not delete any files from your originals folder. This is only possible if you archive images via the user interface, go to the archive and then click on the “Delete all” button. See the documentation I shared with you. Since we are working on new features that our users are waiting for, our core team has very little time to answer questions. I will therefore convert this issue for further discussion – maybe a community member can help with troubleshooting! :)
